src/main/kotlin/org/ptss/support/security/JwtValidator.kt

```diff
@@ -0,0 +1,45 @@
+package org.ptss.support.security
+
+import io.quarkus.security.UnauthorizedException
+import io.smallrye.jwt.auth.principal.JWTParser
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import org.ptss.support.domain.enums.Role
+
+@ApplicationScoped
+class JwtValidator @Inject constructor(
+    private val jwtParser: JWTParser
+) {
+
+    fun validateAccessToken(accessToken: String, allowedRoles: Array<Role>): Role {
+        return try {
+            val claims = jwtParser.parse(accessToken)
+            val tokenRole = mapClaimToRole(claims.getClaim("role") as String?)
+
+            tokenRole.takeIf { it in allowedRoles }
+                ?: throw UnauthorizedException("Insufficient permissions")
+        } catch (e: Exception) {
+            throw UnauthorizedException("Invalid access token")
+        }
+    }
+
+    fun validateRefreshToken(refreshToken: String): Role {
+        return try {
+            val claims = jwtParser.parse(refreshToken)
+            mapClaimToRole(claims.getClaim("role") as String?)
+        } catch (e: Exception) {
+            throw UnauthorizedException("Invalid refresh token")
+        }
+    }
+
+    private fun mapClaimToRole(roleClaim: String?): Role {
+        return when (roleClaim?.uppercase()) {
+            "ADMIN" -> Role.ADMIN
+            "PATIENT" -> Role.PATIENT
+            "HCP" -> Role.HCP
+            "FAMILY_MEMBER" -> Role.FAMILY_MEMBER
+            "PRIMARY_CAREGIVER" -> Role.PRIMARY_CAREGIVER
+            else -> throw UnauthorizedException("Invalid role in token")
+        }
+    }
```

Comment:
Aangezien de naam van de Role enum record en de string exact hetzelfde zijn, is het niet makkelijker om de roleClaim proberen te converten naar een Role instance? Maakt de code ook wat meer dynamisch. Hier is wat example code:
```kotlin
private fun mapClaimToRole(roleClaim: String): Role = logger.executeWithExceptionLoggingAsync(
    operation = { Role.valueOf(roleClaim.uppercase()) },
    logMessage = "Invalid role claim: %s",
    exceptionHandling = { UnauthorizedException("Invalid role claim: $roleClaim") },
    roleClaim
)
```

Verder, waarom kan deze functie als input een nullable String hebben (`String?`). Als de roleClaim nullable is, zal het nooit goed zijn, dat weet je ook voordat je hem probeert te converten naar een Role. 

Als laatste, misschien kan je wel een `fromString` functie aan de Role enum toevoegen, maar dat is een persoonlijke stylistic choice. Het voordeel is dat de functie van overal gebruikt kan worden, maar je wilt dan misschien weer geen logging in the Role enum.